### PR TITLE
Cleanup on destroy

### LINKED ISSUE #8: Remove channels, category and role on !gdestroy
When `!gdestroy` is called the group is deleted. Currently role, text channel, voice channel and category remain while only the group.yml entry is removed. In the future those should be removed automatically as well - if group deletion succeeded.

### PR DESCRIPTION
**Pull request by XaverStiensmeier**
In the past roles, categories and channels remained after group deletion. This is now fixed. This closes #8

### CODE REVIEW

File: cogs/groupsCog.py [hunk 1]
```diff
@@ -51,7 +51,24 @@ async def glist(self, ctx,
...
+        status=True
+        if status:
+            # remove role
+            role = discord.utils.get(ctx.guild.roles, name=group_name)
+            if role:
+                await role.delete()
+
+            # remove category
+            category = discord.utils.get(ctx.guild.categories, name=group_name)
+            if category:
```
Comment (ReneP8) [inline cogs/groupsCog.py:64]: It is helpful for server logs to also log deletion of category.

File: cogs/groupsCog.py [hunk 2]
```diff
@@ -51,7 +51,24 @@ async def glist(self, ctx,
...
         group_name = f"{group_prefix}_{ctx.author}"
         status, result_str = organize_group.destroy_group(group_name)
-        # if exit_status, delete channels and role
+        # if status, delete channels and role
+        status=True
+        if status:
+            # remove role
+            role = discord.utils.get(ctx.guild.roles, name=group_name)
+            if role:
+                await role.delete()
```
Comment (ReneP8) [inline cogs/groupsCog.py:60]: It is helpful for server logs to also log deletion of role.

File: cogs/groupsCog.py (hunk 1 above)
Comment (XaverStiensmeier) [reply cogs/groupsCog.py:64]: What do you mean by that? Do you mean an info log on that? All entered commands are already logged. What would you log additionally?

File: cogs/groupsCog.py (hunk 2 above)
Comment (XaverStiensmeier) [reply cogs/groupsCog.py:60]: What do you mean by that? Do you mean an info log on that? All entered commands are already logged. What would you log additionally?

Comment (XaverStiensmeier) [reply cogs/groupsCog.py:61]: I added some debug logging, which is currently ignored. At a later state one might make debug logging available by using a CL argument or something.

**Pull request merged**